includes/class-wc-payments-order-service.php

```diff
@@ -1431,6 +1466,43 @@ private function generate_payment_failure_note( $intent_id, $charge_id, $message
 
 		return $note;
 	}
+	/**
+	 * Get content for the failure order note and additional message, if included.
+	 *
+	 * @param string   $intent_id        The ID of the intent associated with this order.
+	 * @param string   $charge_id        The charge ID related to the intent/order.
+	 * @param string   $message          Optional message to add to the note.
+	 * @param string   $formatted_amount The formatted order total.
+	 * @param int|null $created          Optional timestamp when the payment was created.
+	 *
+	 * @return string Note content.
+	 */
+	private function generate_terminal_payment_failure_note( $intent_id, $charge_id, $message, $formatted_amount, ?int $created = null ) {
+		$transaction_url = WC_Payments_Utils::compose_transaction_url( $intent_id, $charge_id );
+		$timestamp       = $created ?? time();
+		$formatted_time  = date_i18n( get_option( 'date_format' ) . ' ' . get_option( 'time_format' ), $timestamp );
```

Comment:
The date formats for me as `January 22, 2025 2:25 pm`. What if the second failed transaction is made at the same minute, could that cause a problem?

Reply:
Thank you for pointing this out. With this commit, https://github.com/Automattic/woocommerce-payments/pull/10168/commits/a767d92e8cee44df4874c0123a47b10c3f9e8dbf, I have check that adds seconds to the date, so it should work now.